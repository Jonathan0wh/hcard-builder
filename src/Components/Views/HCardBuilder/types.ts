export interface IFormData {
  givenName: string;
  surname: string;
  email: string;
  tel: string;
  house: string;
  street: string;
  suburb: string;
  state: string;
  postcode: string;
  country: string;
  avatar: File | null;
}

export interface IFormHandler {
  setGivenName: React.Dispatch<React.SetStateAction<string>>;
  setSurname: React.Dispatch<React.SetStateAction<string>>;
  setEmail: React.Dispatch<React.SetStateAction<string>>;
  setTel: React.Dispatch<React.SetStateAction<string>>;
  setHouse: React.Dispatch<React.SetStateAction<string>>;
  setStreet: React.Dispatch<React.SetStateAction<string>>;
  setSuburb: React.Dispatch<React.SetStateAction<string>>;
  setState: React.Dispatch<React.SetStateAction<string>>;
  setPostcode: React.Dispatch<React.SetStateAction<string>>;
  setCountry: React.Dispatch<React.SetStateAction<string>>;
  setAvatar: React.Dispatch<React.SetStateAction<File | null>>;
}
