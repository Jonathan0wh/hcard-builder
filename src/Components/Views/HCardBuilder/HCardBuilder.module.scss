@import './styles/colors';

.container {
  height: 100%;
  background-color: $white;

  :global {
    .ant-row {
      @media only screen and (min-width: 992px) {
        height: 100%;
      }
    }
    .ant-col {
      height: 100%;
    }
  }
}
