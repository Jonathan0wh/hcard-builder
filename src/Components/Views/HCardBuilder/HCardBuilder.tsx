import React, { useState } from 'react';
import { Layout, Row, Col } from 'antd';

import BuilderForm from 'Components/Views/BuilderForm';
import BuilderPreview from 'Components/Views/BuilderPreview';
import { IFormData, IFormHandler } from './types';

import styles from './HCardBuilder.module.scss';

const HCardBuilder = () => {
  const [givenName, setGivenName] = useState('');
  const [surname, setSurname] = useState('');
  const [email, setEmail] = useState('');
  const [tel, setTel] = useState('');
  const [house, setHouse] = useState('');
  const [street, setStreet] = useState('');
  const [suburb, setSuburb] = useState('');
  const [state, setState] = useState('');
  const [postcode, setPostcode] = useState('');
  const [country, setCountry] = useState('');
  const [avatar, setAvatar] = useState();

  const formData: IFormData = {
    givenName,
    surname,
    email,
    tel,
    house,
    street,
    suburb,
    state,
    postcode,
    country,
    avatar
  };
  const formHandler: IFormHandler = {
    setGivenName,
    setSurname,
    setEmail,
    setTel,
    setHouse,
    setStreet,
    setSuburb,
    setState,
    setPostcode,
    setCountry,
    setAvatar
  };

  const { Content } = Layout;

  return (
    <Layout className={styles.container}>
      <Content>
        <Row>
          <Col lg={12}>
            <BuilderForm formData={formData} formHandler={formHandler} />
          </Col>
          <Col lg={12}>
            <BuilderPreview formData={formData} />
          </Col>
        </Row>
      </Content>
    </Layout>
  );
};

export default HCardBuilder;
