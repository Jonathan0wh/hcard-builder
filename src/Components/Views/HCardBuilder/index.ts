export { default } from './HCardBuilder';
export * from './types';
