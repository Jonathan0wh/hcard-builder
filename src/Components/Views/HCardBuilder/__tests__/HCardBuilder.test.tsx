import React from 'react';
import { shallow } from 'enzyme';

import HCardBuilder from '../HCardBuilder';

describe('<HCardBuilder />', () => {
  it('should render by default', () => {
    const wrapper = shallow(<HCardBuilder />);
    expect(wrapper).toMatchSnapshot();
  });
});
