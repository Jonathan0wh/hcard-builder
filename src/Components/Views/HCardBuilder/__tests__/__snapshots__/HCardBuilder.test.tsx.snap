// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HCardBuilder /> should render by default 1`] = `
<Adapter
  className="container"
>
  <Adapter>
    <Row
      gutter={0}
    >
      <Col
        lg={12}
      >
        <BuilderForm
          formData={
            Object {
              "avatar": undefined,
              "country": "",
              "email": "",
              "givenName": "",
              "house": "",
              "postcode": "",
              "state": "",
              "street": "",
              "suburb": "",
              "surname": "",
              "tel": "",
            }
          }
          formHandler={
            Object {
              "setAvatar": [Function],
              "setCountry": [Function],
              "setEmail": [Function],
              "setGivenName": [Function],
              "setHouse": [Function],
              "setPostcode": [Function],
              "setState": [Function],
              "setStreet": [Function],
              "setSuburb": [Function],
              "setSurname": [Function],
              "setTel": [Function],
            }
          }
        />
      </Col>
      <Col
        lg={12}
      >
        <BuilderPreview
          formData={
            Object {
              "avatar": undefined,
              "country": "",
              "email": "",
              "givenName": "",
              "house": "",
              "postcode": "",
              "state": "",
              "street": "",
              "suburb": "",
              "surname": "",
              "tel": "",
            }
          }
        />
      </Col>
    </Row>
  </Adapter>
</Adapter>
`;
